plugin panics when the architecture is ARM64

Hi, 

Thanks once more for this plugin. 

I 've been trying out the plugin with some arm VMs and I got bitten by the following:

```
goroutine 385 [running]:
github.com/k0sproject/k0sctl/phase.(*PrepareArm).etcdUnsupportedArch(0xc000368280?, 0xc0006d4e00)
	github.com/k0sproject/k0sctl@v0.15.2/phase/arm_prepare.go:48 +0xce
github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster.Hosts.ParallelEach.func1(0xc0006d4e00)
	github.com/k0sproject/k0sctl@v0.15.2/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster/hosts.go:115 +0xc4
created by github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster.Hosts.ParallelEach
	github.com/k0sproject/k0sctl@v0.15.2/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster/hosts.go:114 +0x24e
panic: assignment to entry in nil map
```

Looking a bit at [the code in question](https://github.com/k0sproject/k0sctl/blob/main/phase/arm_prepare.go#L48) and trying to reproduce using the k0sctl binary as is, I am a bit at a loss. I can't reproduce it running the k0sctl, that one works fine. 

The code suggests to me that what's missing is the `Environment` map (thus being nil) but I am not yet sure why vanilla k0sctl calls have it and the plugin doesn't.

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,2 @@
 dist/
+!.vscode/

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -1,0 +1,13 @@
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Debug Terraform Provider",
+      "type": "go",
+      "request": "launch",
+      "mode": "debug",
+      "program": "${workspaceFolder}",
+      "args": ["-debug"]
+    }
+  ]
+}

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/alessiodionisi/terraform-provider-k0s
 
-go 1.20
+go 1.21
 
 require (
 	github.com/hashicorp/terraform-plugin-docs v0.16.0

--- a/internal/provider/cluster_resource.go
+++ b/internal/provider/cluster_resource.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
@@ -177,12 +178,14 @@ func (r *ClusterResource) Create(ctx context.Context, req resource.CreateRequest
 	var data *ClusterResourceModel
 
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	k0sctlConfig := getK0sctlConfig(data)
+	k0sctlConfig := getK0sctlConfig(ctx, &resp.Diagnostics, data)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	if err := k0sctlConfig.Validate(); err != nil {
 		resp.Diagnostics.AddError("k0sctl Error", fmt.Sprintf("Unable to create cluster, got error: %s", err))
@@ -206,12 +209,14 @@ func (r *ClusterResource) Read(ctx context.Context, req resource.ReadRequest, re
 	var data *ClusterResourceModel
 
 	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	k0sctlConfig := getK0sctlConfig(data)
+	k0sctlConfig := getK0sctlConfig(ctx, &resp.Diagnostics, data)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	if err := k0sctlConfig.Validate(); err != nil {
 		resp.Diagnostics.AddError("k0sctl Error", fmt.Sprintf("Unable to read cluster, got error: %s", err))
@@ -247,12 +252,14 @@ func (r *ClusterResource) Update(ctx context.Context, req resource.UpdateRequest
 	var data *ClusterResourceModel
 
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	k0sctlConfig := getK0sctlConfig(data)
+	k0sctlConfig := getK0sctlConfig(ctx, &resp.Diagnostics, data)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	if err := k0sctlConfig.Validate(); err != nil {
 		resp.Diagnostics.AddError("k0sctl Error", fmt.Sprintf("Unable to update cluster, got error: %s", err))
@@ -275,12 +282,14 @@ func (r *ClusterResource) Delete(ctx context.Context, req resource.DeleteRequest
 	var data *ClusterResourceModel
 
 	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	k0sctlConfig := getK0sctlConfig(data)
+	k0sctlConfig := getK0sctlConfig(ctx, &resp.Diagnostics, data)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
 	if err := k0sctlConfig.Validate(); err != nil {
 		resp.Diagnostics.AddError("k0sctl Error", fmt.Sprintf("Unable to delete cluster, got error: %s", err))
@@ -377,15 +386,25 @@ func getK0sctlManagerForCreateOrUpdate(data *ClusterResourceModel, k0sctlConfig 
 	return manager
 }
 
-func getK0sctlConfig(data *ClusterResourceModel) *k0sctl_v1beta1.Cluster {
+func getK0sctlConfig(ctx context.Context, dia *diag.Diagnostics, data *ClusterResourceModel) *k0sctl_v1beta1.Cluster {
 	k0sctlHosts := []*k0sctl_cluster.Host{}
 
 	for _, host := range data.Hosts {
 		var installFlags []string
-		host.InstallFlags.ElementsAs(context.Background(), &installFlags, false)
+		dia.Append(host.InstallFlags.ElementsAs(ctx, &installFlags, false)...)
+		if dia.HasError() {
+			return nil
+		}
 
 		var environment map[string]string
-		host.Environment.ElementsAs(context.Background(), &environment, false)
+		dia.Append(host.Environment.ElementsAs(ctx, &environment, false)...)
+		if dia.HasError() {
+			return nil
+		}
+
+		if environment == nil {
+			environment = map[string]string{}
+		}
 
 		k0sctlHosts = append(k0sctlHosts, &k0sctl_cluster.Host{
 			Connection: k0s_rig.Connection{

